HPCC-19746 Copy resource parameters instead of merging

timothyklemm opened the pull request:
Change CSecurityResource::copy to create a copy of the source
resource's parameter array. It currently adds these parameters
to whatever exists in the destination, essentially merging the
results. This violates the semantics of copy, and deviates from
the implementation found in CLdapSecuResource.

Fixed HPCC-19746

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [x] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

timothyklemm commented:
@RussWhitehead Please review this change to CSecurityResource.

RussWhitehead reviewed (changes requested):
The code looks good but the commit is not properly signed. GitGui can create it for you, but should look something like:

Signed-off-by: Tim Klemm tim.klemm@lexisnexis.com

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 3a38252c4076a80e77362c435e99f6978f299675
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |786 |786 |0 |
|test (thor) |717 |717 |0 |
|test (roxie) |830 |830 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   24 sec (00:00:24) |  171 sec (00:02:51) |   60 sec (00:01:00) |    9 sec (00:00:09) |   30 sec (00:00:30) | 1375 sec (00:22:55) |   17 sec (00:00:17) | 1686 sec (00:28:06) |

Commit:
HPCC-19746 Copy resource parameters instead of merging

Change CSecurityResource::copy to create a copy of the source
resource's parameter array. It currently adds these parameters
to whatever exists in the destination, essentially merging the
results. This violates the semantics of copy, and deviates from
the implementation found in CLdapSecuResource.

Fixed HPCC-19746

Signed-off-by: Tim Klemm <Tim.Klemm@lexisnexisrisk.com>

timothyklemm commented:
@RussWhitehead I amended the commit message.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 0f4bc134573ff6a14246a54087992ae4387c3063
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |786 |786 |0 |
|test (thor) |717 |717 |0 |
|test (roxie) |830 |830 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   18 sec (00:00:18) |  173 sec (00:02:53) |   60 sec (00:01:00) |    9 sec (00:00:09) |   30 sec (00:00:30) | 1363 sec (00:22:43) |   18 sec (00:00:18) | 1671 sec (00:27:51) |

RussWhitehead commented:
@timothyklemm Looks good

timothyklemm commented:
@richardkchapman Please merge. This request was created before my account names were synchronized, so I don't think I can link to Jira (I've added a comment).

richardkchapman commented:
@timothyklemm Please rebase for a clean merge

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: e696f1c7e2c0bfe397165fc8baebaef5729f7c26
Build: success
Install hpccsystems-platform-community_7.0.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |89 |89 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |787 |787 |0 |
|test (thor) |718 |718 |0 |
|test (roxie) |831 |831 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   35 sec (00:00:35) |  172 sec (00:02:52) |   61 sec (00:01:01) |    9 sec (00:00:09) |   30 sec (00:00:30) | 1165 sec (00:19:25) |   19 sec (00:00:19) | 1491 sec (00:24:51) |

timothyklemm commented:
@richardkchapman Rebase completed.
